### ISSUE TITLE
`npx pnpm` fails on Windows if home directory contains space

### ISSUE THREAD

**Issue by joheriks** [vaadin/flow#10084]
As  `Windows User` on Windows with home directory `C:\Users\Windows User`, when Flow is fetching a compatible pnpm version via `npx` (using a command prefixed by `npx pnpm ...`) it fails with error `EPERM: operation not permitted, mkdir 'C:\Users\Windows'`. This seems to be due to an issue with npm 6 not accepting spaces in the home directory:
https://stackoverflow.com/questions/55450386/npm-eperm-mkdir-operation-not-permitted-on-windows

**Note: fixed in npm 7 (node.js 15+) (EDIT:) which is installed by default by Flow 8.0 (V21)**. Installing Node.js 15+ globally fixes the issue.

Versions:
- Vaadin 19.0.0.beta3
- Windows 10
- npm 6.14.8 
- node v14.15.5

**Comment by johannest** [comment]
Force upgrading to the latest npm 7 through PowerShell with following commands (https://stackoverflow.com/questions/18412129/how-can-i-update-npm-on-windows) helped me:
```
Set-ExecutionPolicy Unrestricted -Scope CurrentUser -Force
npm install -g npm-windows-upgrade
npm-windows-upgrade
```

**Comment by johannest** [comment]
In case the issue manifests itself with `Found too old 'pnpm'. If installed into the project 'node_modules', upgrade 'pnpm' to at least 5.0.0`

Then the above workaround is not enough, but you might need to 
1. Run `npm config set cache`
2. And run `npm config set cache`

**Comment by pleku** [comment]
Because Vaadin 20 is still coming with Node.js 14 + npm 6 and we are not going to update that on the fly for Vaadin 20, we should probably do something about this for Vaadin 20. There is another issue automatically upgrading the node.js version when updating Vaadin versions #9864.

Thus it might make sense in the meanwhile to 
a) detect windows + space in the home folder
b) check that npm version 6 or older
And fail with a clear message that tells them to update their node.js version and why it should be updated.

**Comment by Haprog** [comment]
> And fail with a clear message that tells them to update their node.js version and why it should be updated.

I don't think you need to update Node.js. Updating `npm`  should be enough.

**Comment by johannest** [comment]
>And fail with a clear message that tells them to update their node.js version and why it should be updated.

It is unfortunately not enough to upgrade the node version. I have v14.7.0, and this is failing to 
`java.lang.IllegalStateException: Found too old 'pnpm'. If installed into the project 'node_modules', upgrade 'pnpm' to at least 5.0.0
        at com.vaadin.flow.server.frontend.FrontendTools.lambda$getSuitablePnpm$1(FrontendTools.java:676) ~[flow-server-8.0.0.alpha10.jar:8.0.0.alpha10]`

**Comment by johannest** [comment]
Neither manually upgrading the npm to 7.19.0 (https://stackoverflow.com/questions/18412129/how-can-i-update-npm-on-windows) does not help anymore.

**Comment by Haprog** [comment]
Hmm https://docs.npmjs.com/try-the-latest-stable-version-of-npm#upgrading-on-windows 👀

**Comment by johannest** [comment]
Finally found workaround. 
1. Make sure Node.js is at least 14.7
2. Manually upgrade npm globally to the latest with Windows PowerShell (https://stackoverflow.com/questions/18412129/how-can-i-update-npm-on-windows)
3. Manually install npx: `npm install -g npx`
4. Manually install pnpm: `npm install -g pnpm`
5. If old project do "Vaadin dance" before starting
